src/TropicalGeometry/hypersurface.jl

```diff
@@ -260,3 +260,62 @@ end
 #     @req has_attribute(TropH,:tropical_semiring_map) "no tropical semiring map cached"
 #     return get_attribute(TropH,:tropical_semiring_map)
 # end
+#
+
+@doc raw"""
+   tropical_prevariety(convention::minmax, A)
+
+Return the tropical prevariety generated by intersecting tropical hypersurfaces corresponding to elements of `A`.
+
+If `A` is a finite collection of polynomials with rational coefficients, return the tropical prevariety obtained by tropicalizing elements of `A` and intersecting the associated hypersurfaces.
+
+The optional input `min` or `max` determines tropicalization convention. Defaults to `min`.
+
+If `A` is a collection of tropical polynomials, the function computes and intersects the associated hypersurfaces.
+# Example
+We compute the Dressian $\text{Dr}(2,5)$ below.
+```jldoctest
+julia> Gr25 = grassmann_pluecker_ideal(2,5)
+Ideal generated by
+  x[[1, 2]]*x[[3, 4]] - x[[1, 3]]*x[[2, 4]] + x[[1, 4]]*x[[2, 3]]
+  x[[1, 2]]*x[[3, 5]] - x[[1, 3]]*x[[2, 5]] + x[[1, 5]]*x[[2, 3]]
+  x[[1, 2]]*x[[4, 5]] - x[[1, 4]]*x[[2, 5]] + x[[1, 5]]*x[[2, 4]]
+  x[[1, 3]]*x[[4, 5]] - x[[1, 4]]*x[[3, 5]] + x[[1, 5]]*x[[3, 4]]
+  x[[2, 3]]*x[[4, 5]] - x[[2, 4]]*x[[3, 5]] + x[[2, 5]]*x[[3, 4]]
+
+julia> Dr25 = tropical_prevariety(gens(Gr25))
+Polyhedral complex in ambient dimension 10
+
+julia> rays_modulo_lineality(Dr25)[1]
+10-element SubObjectIterator{RayVector{QQFieldElem}}:
+ [1, -1//3, -1//3, -1//3, -1//3, -1//3, -1//3, 1//3, 1//3, 1//3]
+ [1, 1, -1, -1, 1, -1, -1, -1, -1, 3]
+ [1, -1, 1, -1, -1, 1, -1, -1, 3, -1]
+ [1, -1, -1, 1, -1, -1, 1, 3, -1, -1]
+ [-1, 1, 1, -1, -1, -1, 3, 1, -1, -1]
+ [-1, -1, -1, 3, 1, 1, -1, 1, -1, -1]
+ [-1, 1, -1, 1, -1, 3, -1, -1, 1, -1]
+ [-1, 3, -1, -1, -1, 1, 1, -1, -1, 1]
+ [-1, -1, 3, -1, 1, -1, 1, -1, 1, -1]
+ [-1, -1, 1, 1, 3, -1, -1, -1, -1, 1]
+```
+"""
+#Input array of polynomials with rational coefficients
+function tropical_prevariety(convention::minmax, A::Vector{<:MPolyRingElem}) where {minmax<:Union{typeof(min),typeof(max)}}
+    length(A) > 0 || error("Empty array.")
+    nu = tropical_semiring_map(coefficient_ring(A[1]), convention)
+    ftrops = [tropical_polynomial(f, nu) for f in A]
+    HTA = [tropical_hypersurface(f) for f in ftrops]
+    return reduce(common_refinement, [h.polyhedralComplex for h in HTA])
+end
+
+#Input array of tropical polynomials
+function tropical_prevariety(F::Vector{TropicalHypersurface{minmax, true}}) where {minmax<:Union{typeof(min),typeof(max)}}
+	length(F) > 0 || error("Empty array.")
+	HTF = [tropical_hypersurface(f) for f in F]
+	return reduce(common_refinement, [h.polyhedralComplex for h in HTF])
+end
+
+
+#Default to min convention
+tropical_prevariety(A::Vector{<:MPolyRingElem}) = tropical_prevariety(min, A)
```

Comment:
```suggestion
function tropical_prevariety(F::Vector{<:MPolyRingElem}, nu::TropicalSemiringMap=tropical_semiring_map(coefficient_ring(first(F))))
    return tropical_prevariety(tropical_polynomial.(F,Ref(nu))
end
```

I suggest adding a second optional `TropicalSemiringMap` that encodes a valuation + convention (default: trivial valuation, min convention).  This allows users working over fields to create tropical hypersurfaces with respect to non-trivial valuations.